job/deploy_manager.go

```diff
@@ -121,17 +121,17 @@ func (m *deployManager) Assign() {
 		return
 	}
 
-	for int(atomic.LoadInt32(&m.deployerCapacity)) > 0 {
-		// TODO: avoid having multiple query
-		jobDeployment, err := m.deployRepository.GetFirstExecutableRequest(ctx)
-		if err != nil {
-			if errors.Is(err, store.ErrResourceNotFound) {
-				m.l.Debug("no deployment request found to assign deployer")
-				return
-			}
-			m.l.Error("failed to fetch executable deployment request to assign deployer", "error", err.Error())
+	count := int(atomic.LoadInt32(&m.deployerCapacity))
+	jobDeployments, err := m.deployRepository.GetExecutableRequest(ctx, count)
+	if err != nil {
+		if errors.Is(err, store.ErrResourceNotFound) {
```

Comment:
since we will get multiple requests (hopefully), will we ever meet this error? I believe checking whether the return is empty or not could be considered.

Reply:
checking this, err resource not found is never met. updated